### PR TITLE
Require whitespace before description with limited start tokens

### LINKED ISSUE #125: Unnamed vargs type in callable/Closure phpdoc is silently ignored
# Bug report

Currently

```
@return \Closure(int ...$u, string): string
```

phpdoc works correctly, but

```
@return \Closure(...int, object): string
```

is silently ignored. No wrong phpdoc error, no error when this type does not match the type of the returned value.

Even if such type is not documented as supported - https://phpstan.org/writing-php-code/phpdoc-types#callables - I belive it should be supported as standard (non-vargs) unnamed parameters are supported.

### Code snippet that reproduces the problem

https://phpstan.org/r/f3779e03-f24f-43e7-8d73-b27973bbd15c

### Expected output

The 2nd phpdoc must emit a phpstan error, the type (return type of the `Closure`) does not match the result.

### PR DESCRIPTION
**Pull request by rvanvelzen**
Fixes #125

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
I really like this! I'm gonna check the integration tests in PHPStan before tagging a release here because there might be some real-world impact.

**Comment by ondrejmirtes** [comment]
Thank you!

File: tests/PHPStan/Parser/PhpDocParserTest.php [hunk 1]
```diff
@@ -1310,6 +1310,58 @@ public function provideReturnTagsData(): Iterator
...
+		];
+
+		yield [
+			'invalid variadic callable',
+			'/** @return \Closure(...int, string): string */',
+			new PhpDocNode([
+				new PhpDocTagNode(
+					'@return',
+					new InvalidTagValueNode(
+						'\Closure(...int, string): string',
```
Comment (mvorisek) [inline tests/PHPStan/Parser/PhpDocParserTest.php:1354]: thank you for throwing at least

as all other types in `\Closure` phpdoc does not require a variable name, can support for unnamed variadic phpdoc be added?

Comment (rvanvelzen) [reply tests/PHPStan/Parser/PhpDocParserTest.php:1354]: It is already supported, but the type always has to come first.

Comment (mvorisek) [reply tests/PHPStan/Parser/PhpDocParserTest.php:1354]: I didn't know, thank you ❤️

**Pull request merged**